### DISCUSSION TITLE
Error while trying to use z3 in jupyter

### DISCUSSION THREAD

**Discussion by Navas73** [Z3Prover/z3#5610] [Q&A]
so, I tried to import z3 and this happened:

"AttributeError: function 'Z3_solver_propagate_init' not found"

before this my error was that it couldn't find libz3.dll, I managed to fix it but maybe I installed an older file? not sure

could not find a solution anywhere. any help?

thanks

**Comment by NikolajBjorner** [comment]
right, jupyter is pointing to a stale libz3.dll.
I use "pip install z3-solver" and "pip install -U ..." for everything jupyter and it tends to handle binary dependencies and versions correctly.
